src/scf/direct_minimization.jl

```diff
@@ -34,19 +34,20 @@ struct DMPreconditioner
     Nk::Int
     Pks::Vector # Pks[ik] is the preconditioner for k-point ik
     unpack::Function
+    kweights::Vector
 end
 function LinearAlgebra.ldiv!(p, P::DMPreconditioner, d)
     p_unpack = P.unpack(p)
     d_unpack = P.unpack(d)
     for ik = 1:P.Nk
-        ldiv!(p_unpack[ik], P.Pks[ik], d_unpack[ik])
+        ldiv!(p_unpack[ik], P.Pks[ik], ((P.kweights[ik])^-1)*d_unpack[ik])
```

Comment:
To avoid allocating memory here can you do:
```suggestion
        ldiv!(p_unpack[ik], P.Pks[ik], d_unpack[ik])
        ldiv!(P.kweights[ik], p_unpack[ik])
```

Reply:
Smart! Shall i add that to the PR or something? (I'm not well versed in contributing to professional public repos😅) I look forward to see how all the changes may impact the performance of `direct_minimization()` on your experiments.

Reply:
Yes please !